### PR TITLE
Add devcontainers

### PR DESCRIPTION
**Pull request by jrhemstad**

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    "SCCACHE_BUCKET": "rapids-sccache-devs",
+    "VAULT_HOST": "https://vault.ops.k8s.rapids.ai",
+    "HISTFILE": "${containerWorkspaceFolder}/.cache/._bash_history"
+  },
+
+  "mounts": [
+    "source=${localWorkspaceFolder}/.cache/.aws,target=/home/coder/.aws,type=bind,consistency=consistent",
+    "source=${localWorkspaceFolder}/.cache/.config,target=/home/coder/.config,type=bind,consistency=consistent"
+  ]
+}
```
Comment (miscco) [inline .devcontainer/devcontainer.json:20]:
```suggestion
}

```

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+
+{
+  "image": "rapidsai/devcontainers:23.06-cpp-gcc12-cuda12.1-ubuntu22.04",
+
+  "hostRequirements": { "gpu": true },
+
+  "initializeCommand": ["/bin/bash", "-c", "mkdir -p .cache/.{aws,config}"],
```
Comment (miscco) [inline .devcontainer/devcontainer.json:7]: We should be consistent

```suggestion
  "initializeCommand": [
    "/bin/bash",
    "-c",
    "mkdir -p .cache/.{aws,config}"
  ],
```

**Review by miscco — APPROVED** [review]
I am wondering, is there a way to derive from a devcontainer template, almost everything here is duplicated biolerplate

**Comment by jrhemstad** [comment]
> I am wondering, is there a way to derive from a devcontainer template, almost everything here is duplicated biolerplate

Not that I'm aware of, that's why I wrote the script to automatically generate all of them from the matrix.

**Pull request merged**